src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -111,18 +111,21 @@ public static void SetArrayItemNodes(ExecutionContext context, ArrayExecutionNod
                     var node = BuildExecutionNode(parent, itemType, parent.Field, parent.FieldDefinition, index);
                     node.Result = d;
 
-                    if (node is ObjectExecutionNode objectNode)
-                    {
-                        SetSubFieldNodes(context, objectNode);
-                    }
-                    else if (node is ArrayExecutionNode arrayNode)
-                    {
-                        SetArrayItemNodes(context, arrayNode);
-                    }
-                    else if (node is ValueExecutionNode valueNode)
+                    if (!(d is IDataLoaderResult))
```

Comment:
SetSubFieldNodes and similar was incorrectly executing twice for `IDataLoaderResult` nodes, even if none of the values were `null`.  However, when a value was `null`, the sub-field nodes had already been set when they should not have been, causing a null reference exception.

Reply:
> SetSubFieldNodes and similar was incorrectly executing twice for IDataLoaderResult nodes, even if none of the values were null

What side effects did it have?

Reply:
Other than the bug already noted, performance only.  Ran some code twice when it didn't need to.